src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java

```diff
@@ -12,30 +13,22 @@
  * @author adrianhopebailie
  *
  */
-public class PreimageSha256Fulfillment implements Fulfillment {
+public class PreimageSha256Fulfillment extends FulfillmentBase {
+
+	public static final ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
+
+	public PreimageSha256Fulfillment(ConditionType type, byte[] payload) {
+		super(type, payload);
+	}
 
-	private static ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
-	
 	private static EnumSet<FeatureSuite> BASE_FEATURES = EnumSet.of(
 			FeatureSuite.SHA_256, 
 			FeatureSuite.PREIMAGE
 		);
 
-	private byte[] preimage;
```

Comment:
Droped forced by the immutability of final and the java syntax. In the constructor (unique place to init the preimage) java forces to call first the base constructor. This generated a null pointer since the base constructor tries to generate a condition while the preimage was not yet initialized. This is one of the things that get fixed using a single class implementation with Collaborator classes. Using inheritance force an order of execution that makes things more complex than desired.